src/fswAlgorithms/sensorInterfaces/scanningInstrumentController/scanningInstrumentController.c

```diff
@@ -0,0 +1,107 @@
+{
+    // check if the required message has not been connected
+    if (!AccessMsg_C_isLinked(&configData->accessInMsg)) {
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: scanningInstrumentController.accessInMsg was not connected.");
+    }
+    if (!AttGuidMsg_C_isLinked(&configData->attGuidInMsg)) {
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: scanningInstrumentController.attGuidInMsg was not connected.");
+    }
+}
+
+
+/*! Add a description of what this main Update() routine does for this module
+ @return void
+ @param configData The configuration data associated with the module
+ @param callTime The clock time at which the function was called (nanoseconds)
+ @param moduleID The module identifier
+*/
+void Update_scanningInstrumentController(scanningInstrumentControllerConfig *configData, uint64_t callTime, int64_t moduleID)
+{
+    double sigma_BR_norm; //!< Norm of sigma_BR
+    double omega_BR_norm; //!< Norm of omega_BR
+
+    AccessMsgPayload accessInMsgBuffer;  //!< local copy of message buffer
+    AttGuidMsgPayload attGuidInMsgBuffer;  //!< local copy of message buffer
+    DeviceStatusMsgPayload deviceStatusInMsgBuffer;  //!< local copy of message buffer
+    DeviceCmdMsgPayload deviceCmdOutMsgBuffer;  //!< local copy of message buffer
+
+    // always zero the output message buffers before assigning values
+    deviceCmdOutMsgBuffer = DeviceCmdMsg_C_zeroMsgPayload();
+
+    // read in the input messages
+    accessInMsgBuffer = AccessMsg_C_read(&configData->accessInMsg);
+    attGuidInMsgBuffer = AttGuidMsg_C_read(&configData->attGuidInMsg);
+
+    // Read in the device status message if it is linked
+    if (DeviceStatusMsg_C_isLinked(&configData->deviceStatusInMsg)) {
+        deviceStatusInMsgBuffer = DeviceStatusMsg_C_read(&configData->deviceStatusInMsg);
+        configData->controllerStatus = deviceStatusInMsgBuffer.deviceStatus;
+    }
+
+    // Compute the norms of the attitude and rate errors
+    sigma_BR_norm = v3Norm(attGuidInMsgBuffer.sigma_BR);
+    omega_BR_norm = v3Norm(attGuidInMsgBuffer.omega_BR_B);
+    
+    // If the controller is active
+    if (configData->controllerStatus) {
+        /* If the attitude error is less than the tolerance, the groundLocation is accessible, and (if enabled) the rate
+        error is less than the tolerance, turn on the instrument and set the imaged indicator to 1*/
+        if ((sigma_BR_norm <= configData->attErrTolerance)
+            && (!configData->useRateTolerance || (omega_BR_norm <= configData->rateErrTolerance)) // Check rate tolerance if useRateTolerance enabled
+            && (accessInMsgBuffer.hasAccess))
+        {
+            deviceCmdOutMsgBuffer.deviceCmd = 1;
+        }
+    }
+
+    // write to the output messages
+    DeviceCmdMsg_C_write(&deviceCmdOutMsgBuffer, &configData->deviceCmdOutMsg, moduleID, callTime);
+
+    return;
```

Comment:
:small_red_triangle_down: This redundant `return` statement can be removed.

Reply:
Redundant return removed.